src/session_impl.cpp

```diff
@@ -558,6 +558,7 @@ namespace aux {
 
 	void session_impl::init(boost::shared_ptr<settings_pack> pack)
 	{
+		INVARIANT_CHECK;
```

Comment:
``init()`` is not a public function, but it should (most likely) establish the invariant at the end of it. I'm a little bit surprised this works..